indra/newview/llfloaterimsessiontab.cpp

```diff
@@ -1038,8 +1076,16 @@ void LLFloaterIMSessionTab::reloadEmptyFloaters()
 
 void LLFloaterIMSessionTab::updateCallBtnState(bool callIsActive)
 {
+    static int prev_state = 0;
+    int new_state = callIsActive ? 1 : 2;
     mVoiceButton->setImageOverlay(callIsActive? getString("call_btn_stop") : getString("call_btn_start"));
     mVoiceButton->setToolTip(callIsActive? getString("end_call_button_tooltip") : getString("start_call_button_tooltip"));
+    mVoiceButtonHangUpMode = callIsActive;
```

Comment:
This seems odd. Voice is in a specific state and button should corelate to that, not try and tracks states on its own. Letting user click ''hang up" multiple times if voice got stuck shutting down or initializing isn't helpfull.

I think instead button should check voice's current state and represent that state accurately.

Reply:
This mode variable just means what exactly user sees on the button,
what he expects from the click - start call or hang up

Reply:
The comment is added to the variable declaration
![image](https://github.com/user-attachments/assets/3bab053d-4bb6-4092-b16a-e4a9d1172a0e)

Reply:
Not what I mean. If `mVoiceButtonHangUpMode` you are setting `enable = TRUE` regardless of voice state. Why are you tracking state independently? Why not check voice's state instead?

Besides won't this break if there are multiple LLFloaterIMSessionTab?

Reply:
Because the voice state can be changed later than we enable/disable the button
The final check should be done after the user clicks the button and callback is executed